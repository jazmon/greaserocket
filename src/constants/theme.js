// @flow
import color from 'color';

const theme = {
  DARK_PRIMARY_COLOR: '#388E3C',
  DEFAULT_PRIMARY_COLOR: '#4CAF50',
  LIGHT_PRIMARY_COLOR: '#C8E6C9',
  TEXT_PRIMARY_COLOR: '#FFFFFF',
  ACCENT_COLOR: '#FFC107',
  PRIMARY_TEXT_COLOR: 'rgba(0, 0, 0, 0.87)',
  SECONDAY_TEXT_COLOR: 'rgba(0, 0, 0, 0.54)',
  DIVIDER_COLOR: 'rgba(0, 0, 0, 0.12)',
  DISABLED_TEXT_COLOR: 'rgba(0, 0, 0, 0.38)',
  WHITE: '#FFF',
  BLACK: '#000',
};

export default {
  ...theme,
  LIGHT_STATUS_BAR: color(theme.DARK_PRIMARY_COLOR).dark(),
};
