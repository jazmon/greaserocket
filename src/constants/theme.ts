// @flow
import color from 'color';

const theme = {
  colors: {
    primary: {
      normal: '#4CAF50',
      dark: '#388E3C',
      light: '#C8E6C9',
    },
    accent: {
      normal: '#FFC107',
    },
    white: '#FFFFFF',
    black: '#000000',
    transparent: 'transparent',
  },
  text: {
    colors: {
      light: '#FFFFFF',
      primary: 'rgba(0, 0, 0, 0.87)',
      secondary: 'rgba(0, 0, 0, 0.54)',
      disabled: 'rgba(0, 0, 0, 0.38)',
      divider: 'rgba(0, 0, 0, 0.12)',
    },
    size: 12,
  },
};

export type ThemeType = typeof theme;

export function getStatusBarTextColor(
  backgroundColor: string = 'light-content',
): 'light-content' | 'dark-content' {
  return color(backgroundColor).dark() ? 'light-content' : 'dark-content';
}

export default theme;
