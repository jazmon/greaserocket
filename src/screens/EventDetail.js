// @flow
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import theme from 'constants/theme';

type Props = {
  navigation: Object,
  route: Object,
  loading: boolean,
};

type State = {
  text: string,
};

class EventDetail extends React.Component {
  props: Props;

  constructor(props: Props) {
    super(props);

    this.state = {
      text: 'EventDetail',
    };
  }

  state: State;

  render() {
    const { text } = this.state;
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{text}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: theme.WHITE,
  },
  text: {
    color: theme.BLACK,
    fontSize: 16,
  },
});

export const EventDetailComponent = EventDetail;
export default EventDetail;
