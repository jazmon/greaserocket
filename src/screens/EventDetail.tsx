// @flow
import * as React from 'react';
import styled from 'styled-components/native';
import { ThemeType } from 'constants/theme';

type Props = {
  navigation: Object;
  route: Object;
  loading: boolean;
};

type State = {
  text: string;
};

const Container = styled.View`
  flex: 1;
  flex-direction: column;
  background-color: ${({ theme }: { theme: ThemeType }) => theme.colors.white};
`;
const BaseText = styled.Text`
  color: ${({ theme }: { theme: ThemeType }) => theme.text.colors.primary};
  font-size: 16px;
`;

class EventDetail extends React.Component<Props, State> {
  props: Props;

  constructor(props: Props) {
    super(props);

    this.state = {
      text: 'EventDetail',
    };
  }

  state: State;

  render() {
    const { text } = this.state;
    return (
      <Container>
        <BaseText>
          {text}
        </BaseText>
      </Container>
    );
  }
}

export const EventDetailComponent = EventDetail;
export default EventDetail;
