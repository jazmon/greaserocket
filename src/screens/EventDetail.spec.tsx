import 'react-native';
import * as React from 'react';
// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from 'constants/theme';

import { EventDetailComponent as EventDetail } from './EventDetail';

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <EventDetail />
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});
