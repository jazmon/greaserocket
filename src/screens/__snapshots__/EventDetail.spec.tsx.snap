// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    EventDetail
  </Text>
</View>
`;
